external/fv3fit/fv3fit/reservoir/reservoir.py

```diff
@@ -0,0 +1,120 @@
+import dataclasses
+import logging
+import numpy as np
+import scipy
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class ReservoirHyperparameters:
+    """Hyperparameters for reservoir
+
+    input_dim: Size of input vector
+    reservoir_state_dim: Size of hidden state vector,
```

Comment:
Could also drop the `reservoir` part here and just call this `state_size`, since it's a bit redundant w/ the `ReservoirHyperparameters` class name.